src/Microsoft.AspNetCore.Routing/Template/TemplateMatcher.cs

```diff
@@ -82,20 +81,25 @@ public bool TryMatch(PathString path, RouteValueDictionary values)
             //
             // On the second pass, we're almost certainly going to match the URL, so go ahead and allocate the 'values'
             // and start capturing strings. 
-            foreach (var requestSegment in pathTokenizer)
+            foreach (var pathSegment in pathTokenizer)
```

Comment:
Why the rename? `requestSegment` seems less ambiguous w.r.t. template versus request.